Add get_module

CodyCBakerPhD opened the pull request:
Given how often we make use of this function in individual conversions, I'd say its a valuable addition to our `conversion_tools`.

luiztauffer reviewed (commented):
I agree this has potential to be re-used several times. Some comments:
- why use `description` instead of `name` to identify the processing module? Then if it doesn't exist we create a module with the name passed as description?
- Should we rename `check_module`, for sake of specificity? Examples: `get_processing_module`, `get_or_create_processing_module`...

CodyCBakerPhD commented:
@luiztauffer Thanks for the review!

1. The `name` is always used to identify the module; the `description` is checked against the module `description` if it exists (and if differs, throws warning), but I have now changed the default to "No description." in the case when the module is created without a specified `description`.
2. I agree on the name - `check_module` sounds like something that would return a boolean for the existence or non-existence of the module in the nwbfile. I've renamed to `get_module`, since all modules are `ProcessingModules` adding processing seems unnecessary, and the `_or_create` feature is a descriptive property that can be conveyed in the docstring. Suffice to say, this function will _always give_ you a module if you ask to _get_ one, so that ought to be appropriate.

luiztauffer reviewed (approved):
good to go!
